modules/Essentiality.py

```diff
@@ -0,0 +1,370 @@
+
+        # Get a list of essential genes:
+        essential_genes_df = self._prepare_essential_genes(
+            f"{input_folder}/{self.ESSENTIAL_GENE_LIST}"
+        )
+
+        # Get a list of effect for each gene/cell line pair:
+        gene_effect_df = self._read_and_melt(
+            f"{input_folder}/{self.GENE_EFFECTS}", "geneEffect"
+        )
+
+        # Read and process expression data:
+        expression_df = self._read_and_melt(
+            f"{input_folder}/{self.GENE_EXPRESSION}", "expression"
+        )
+
+        # Read and process hotspot mutation:
+        hotspots = self._read_and_melt(
+            f"{input_folder}/{self.MUTATION_HOTSPOTS}", "hotspotMutation"
+        ).filter(f.col("hotspotMutation") > 0)
+
+        # Read and process damaging mutation:
+        damaging_mutation = self._read_and_melt(
+            f"{input_folder}/{self.MUTATION_DAMAGING}", "damagingMutation"
+        ).filter(f.col("damagingMutation") > 0)
+
+        # Get tissue mapping:
+        tissue_mapping = self._get_tissue_mapping(tissue_mapping_url)
+
+        # Join all the data together:
+        self.essentials = (
+            # The melted effect table:
+            gene_effect_df
+            # Narrowing down the gene effect table to genes that are in the essential gene list:
+            .join(essential_genes_df, on="targetSymbol", how="left")
+            .repartition("depmapId")
+            # Joining with model information:
+            .join(models_df, on="depmapId", how="left")
+            # Joining expression data:
+            .join(expression_df, on=["targetSymbol", "depmapId"], how="left")
+            # Joining damaging mutation data:
+            .join(damaging_mutation, on=["targetSymbol", "depmapId"], how="left")
+            # Joining hotspot mutation data:
+            .join(hotspots, on=["targetSymbol", "depmapId"], how="left")
+            # Joining tissue mapping:
+            .join(tissue_mapping, on="tissueFromSource", how="left")
+            # Format data:
+            .select(
+                "targetSymbol",
+                "depmapId",
+                # Cell-line fields:
+                "cellLineName",
+                f.col("modelId").alias("diseaseCellLineId"),
+                # Disease:
+                "diseaseFromSource",
+                # Tissue fields:
+                "tissueId",
+                f.when(f.col("tissueName").isNull(), "other")
+                .otherwise(f.col("tissueName"))
+                .alias("tissueName"),
+                # Parsing mutation:
+                f.when(f.col("damagingMutation").isNotNull(), "damaging")
+                .when(f.col("hotspotMutation").isNotNull(), "hotspot")
+                .alias("mutation"),
+                # Measured values:
+                "geneEffect",
+                "expression",
+                # Essentiality flag:
+                f.when(f.col("isEssential"), True)
```

Comment:
You don't need a when/otherwise clause here. You can do `(f.col("isEssential") == True)` and it will return a boolean.

Reply:
It doesn't work like that. If `f.col("isEssential")` is null, the test `f.col("isEssential") == True` also returns null. That condition was meant to account for this.

```python
(
    spark.createDataFrame([
        ('a', True),
        ('a', None),
    ], ['str', 'bool'])
    .select('str',(f.col('bool') == True).alias('bool'))
    .show()
)
```

```
+---+----+
|str|bool|
+---+----+
|  a|true|
|  a|null|
+---+----+
```

However, this works:

```
 .select('str',f.coalesce(f.col('bool'), f.lit(False)).alias('bool'))
```